// Generated by the code_gen tool.  DO NOT EDIT!
// source file: /home/kevin/SVN/project/mdl/src/resource/mdl_szl1_msg.xml
// time: 2015/07/18 14:14:24

package datayes.mdl.mdl_szl1_pbmsg;
option java_outer_classname = "MDLSZL1Msg";

enum MDLSZL1ServiceVersion {
	MDLVID_MDL_SZL1 = 101; 
};

enum MDLSZL1MessageID {
	MDLMID_MDL_SZL1_SZL1Index = 1;
	MDLMID_MDL_SZL1_SZL1Stock = 2;
};

message double_3 {
	required int64 Value =  1;
	optional uint32 DecimalShift = 2 [default = 1000];
};

message float_3 {
	required int32 Value =  1;
	optional uint32 DecimalShift = 2 [default = 1000];
};

message SZL1Index {
	enum MID {
		MessageID = 1;
	};
	enum SID {
		ServiceID = 5;
	};
	enum VID {
		ServiceVersion = 101;
	};
	optional string IndexID = 1;
	optional string IndexName = 2;
	optional double_3 PreCloseIndex = 3;
	optional double_3 OpenIndex = 4;
	optional double_3 HighIndex = 5;
	optional double_3 LowIndex = 6;
	optional double_3 LastIndex = 7;
	optional double_3 TradVolume = 8;
	optional double_3 Turnover = 9;
	optional uint32 UpdateTime = 10;
};

message SZL1Stock {
	enum MID {
		MessageID = 2;
	};
	enum SID {
		ServiceID = 5;
	};
	enum VID {
		ServiceVersion = 101;
	};
	optional string SecurityID = 1;
	optional string SecurityName = 2;
	optional float_3 PreCloPrice = 3;
	optional float_3 OpenPrice = 4;
	optional double_3 Turnover = 5;
	optional uint32 TurnNum = 6;
	optional float_3 HighPrice = 7;
	optional float_3 LowPrice = 8;
	optional float_3 LastPrice = 9;
	optional float_3 PE1 = 10;
	optional float_3 PE2 = 11;
	optional float_3 AskPrice5 = 12;
	optional uint64 AskVolume5 = 13;
	optional float_3 AskPrice4 = 14;
	optional uint64 AskVolume4 = 15;
	optional float_3 AskPrice3 = 16;
	optional uint64 AskVolume3 = 17;
	optional float_3 AskPrice2 = 18;
	optional uint64 AskVolume2 = 19;
	optional float_3 AskPrice1 = 20;
	optional uint64 AskVolume1 = 21;
	optional float_3 BidPrice1 = 22;
	optional uint64 BidVolume1 = 23;
	optional float_3 BidPrice2 = 24;
	optional uint64 BidVolume2 = 25;
	optional float_3 BidPrice3 = 26;
	optional uint64 BidVolume3 = 27;
	optional float_3 BidPrice4 = 28;
	optional uint64 BidVolume4 = 29;
	optional float_3 BidPrice5 = 30;
	optional uint64 BidVolume5 = 31;
	optional uint32 UpdateTime = 32;
	optional float_3 DifPrice1 = 33;
	optional float_3 DifPrice2 = 34;
	optional uint64 OpenInt = 35;
	optional uint64 Volume = 36;
	optional uint32 DeletionIndicator = 37;
};
